src/bin/miri.rs

```diff
@@ -236,6 +236,12 @@ fn main() {
                 "-Zmiri-track-raw-pointers" => {
                     miri_config.track_raw = true;
                 }
+                "-Zmiri-track-dummy-op" => {
+                    miri_config.dummy_op_visibility = miri::DummyOpVisibility::Track;
+                }
+                "-Zmiri-ignore-dummy-op" => {
+                    miri_config.dummy_op_visibility = miri::DummyOpVisibility::Hide;
+                }
```

Comment:
I don't think we should have two flags for this... also, "dummy op" is fairly unclear as a name.

I propose that instead we have a single flag, srawman name: `-Zmiri-isolated-op=<hide|warn|track>`. (I am not sure if "warn" vs "track" is a very meaningful difference here, I think it should be "warn" no matter the underlying implementation.) The underlying enum should probably also have a variant "allow" which corresponds to disabling isolation.

Reply:
> Looks like NonHaltingDiagnostic uses the same diagnostic infra, which dedups the messages. So it shows a single message per line in src. But because it's too verbose to print the backtrace by default, I pushed a change to throw a warning without a span or backtrace.

So maybe the values should be `<hide|warn|warn-nobacktrace|allow>`? (where the latter is equivalent to `-Zmiri-disable-isolation`)

Reply:
I used "dummy op" to represent a subset of ops which are faked in isolation. I'm fine with `-Zmiri-isolated-op`. 

> I think it should be "warn" no matter the underlying implementation

Do you mean to give user only two options - <hide|warn>, where `warn` prints backtraces, and without `-Zmiri-isolated-op` (by default), miri prints just a warning?

> The underlying enum should probably also have a variant "allow" which corresponds to disabling isolation.

Sorry, I didn't get this part. Presumably this variant controls isolation for such foreign ops (i.e. miri executes the op even in isolation), how does it get set? We provide one more option in `-Zmiri-isolated-op`? How is it different from `-Zmiri-disable-isolation`?

Reply:
I somehow missed your second comment, which clarifies my previous questions.
`-Zmiri-isolated-op=<hide|warn|warn-nobacktrace|allow>`, with `warn-nobacktrace` as default, sounds good to me. Will update the PR soon.

Reply:
Sorry, by "underlying enum" I meant the field in `MiriConfig`.

`-Zmiri-disable-isolation` would then be syntactic sugar for `-Zmiri-isolated-op=allow`.

> with warn-nobacktrace as default

I am honestly not sure if `warn` or `warn-nobacktrace` should be the default.

Reply:
> `-Zmiri-disable-isolation` would then be syntactic sugar for `-Zmiri-isolated-op=allow`

Is that so? Latter shouldn't change machine level `communicate` flag like the former do, right? I thought, when it's time to execute `chdir`, for example, we check for `machine.communicate || miri-config.isolated-op == allow`?

Reply:
When `communicate` is `true`, then the other isolated-op actions make no sense, right? It's always bad to have "nonesnsical" combinations. That's why I think this should be all one large enum.

What else would `-Zmiri-isolated-op=allow` means, when it is not the same as `-Zmiri-disable-isolation`?

>  we check for machine.communicate || miri-config.isolated-op == allow?

If we do that everywhere the `communicate` flag is used, we might as well remove that flag -- no?

Reply:
Ok, that makes sense. I wasn't sure if `communicate` has semantics other than just controlling isolated ops. I just pushed the change which captures all the settings in a single enum. For now, I've kept the `communicate` flag as it is, as it'd require touching more files. I'm not sure about the preference for smaller PRs. If you like the new changes and prefer to deprecate `communicate` in the same PR, I'd be happy to do 🙂

Reply:
It shouldn't have any other semantics.

I'd prefer not to have redundant option flags in the machine settings, so if you could e.g. replace `communicate` by `fn communicate()` or so, that would be great. :)